### PR TITLE
Codegen for purview scanning

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Code change:
1. llc codegen for scanning
2. update/rewrite samples and tests, pass live test and playback test
3. update session-records
4. update sdk/purview/azure-analytics-purview-scanning/pom.xml 
5. update sdk/purview/azure-analytics-purview-scanning/CHANGELOG.md
6. update sdk/purview/azure-analytics-purview-scanning/README.md

### CODE REVIEW

File: sdk/purview/azure-analytics-purview-scanning/src/test/java/com/azure/analytics/purview/scanning/SystemScanRulesetsBaseClientTests.java [hunk 1]
```diff
@@ -3,39 +3,31 @@
...
-import javax.json.JsonObject;
-import javax.json.JsonReader;
-import java.io.StringReader;
+import java.util.List;
+import java.util.stream.Collectors;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class SystemScanRulesetsBaseClientTests extends PurviewScanningClientTestBase {
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-scanning/src/test/java/com/azure/analytics/purview/scanning/SystemScanRulesetsBaseClientTests.java:15]: Please rename the class to e.g. `SystemScanRulesetsClientTests`

File: sdk/purview/azure-analytics-purview-scanning/README.md [hunk 2]
```diff
@@ -50,19 +50,19 @@ To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below
 <dependency>
     <groupId>com.azure</groupId>
     <artifactId>azure-identity</artifactId>
-    <version>1.2.5</version>
+    <version>1.3.6</version>
 </dependency>
 ```
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-scanning/README.md:55]: misses `[//]: # ({x-version-update-end})`

File: sdk/purview/azure-analytics-purview-scanning/pom.xml [hunk 3]
```diff
@@ -40,12 +40,7 @@
...
       <artifactId>azure-core</artifactId>
-      <version>1.20.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
-    </dependency>
-    <dependency>
-      <groupId>com.azure</groupId>
-      <artifactId>azure-core-experimental</artifactId>
-      <version>1.0.0-beta.18</version> <!-- {x-version-update;com.azure:azure-core-experimental;dependency} -->
+      <version>1.21.0-beta.1</version> <!-- {x-version-update;beta_com.azure:azure-core;dependency} -->
     </dependency>
     <dependency>
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-scanning/pom.xml:45]: clean up unused dependencies e.g. `javax.json`

File: sdk/purview/azure-analytics-purview-scanning/src/samples/java/com/azure/analytics/purview/scanning/ListAllSystemScanRulesets.java [hunk 4]
```diff
@@ -3,14 +3,16 @@
...
 import javax.json.Json;
-import javax.json.JsonArray;
 import javax.json.JsonObject;
 import javax.json.JsonReader;
 import java.io.StringReader;
+import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Sample for listing all system scan rulesets using the SystemScanRulesetsBaseClient.
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-scanning/src/samples/java/com/azure/analytics/purview/scanning/ListAllSystemScanRulesets.java:18]: No longer the `SystemScanRulesetsBaseClient`

File: sdk/purview/azure-analytics-purview-scanning/pom.xml (hunk 3 above)
Comment (haolingdong-msft) [reply sdk/purview/azure-analytics-purview-scanning/pom.xml:45]: We used javax.json in src/samples/java/com/azure/analytics/purview/scanning/ListAllSystemScanRulesets.java, I will clean up other unused ones.

File: sdk/purview/azure-analytics-purview-scanning/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 # Release History
 
+## 1.0.0-beta.2 (Unreleased)
+
+
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-scanning/CHANGELOG.md:5]: https://azure.github.io/azure-sdk/policies_releases.html#change-logs when upon release

**Pull request merged**